sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -69,7 +69,14 @@ private[sql] class HiveSessionCatalog(
     // Current thread context classloader may not be the one loaded the class. Need to switch
     // context classloader to initialize instance properly.
     Utils.withContextClassLoader(clazz.getClassLoader) {
-      Try(super.makeFunctionExpression(name, clazz, input)).getOrElse {
+      val t = Try(super.makeFunctionExpression(name, clazz, input))
+      if (t.isSuccess) {
```

Comment:
I'd prefer using native `try catch` instead of writing `if else` for `Try`.

Reply:
> I'd prefer using native `try catch` instead of writing `if else` for `Try`.

To keep indentation so I change use `try catch` ....  
 https://github.com/apache/spark/pull/29054#discussion_r452812103